src/common/test/componentMocking.tsx

```diff
@@ -17,26 +22,67 @@ import {
 } from '../../profile/context/ProfileContext';
 import {
   MockApolloClientProvider,
+  resetApolloMocks,
   ResponseProvider,
 } from './MockApolloClientProvider';
 import { AnyObject } from '../../graphql/typings';
 import ToastProvider from '../../toast/__mocks__/ToastProvider';
+import {
+  MutationReturnType,
+  useProfileMutations,
+} from '../../profile/hooks/useProfileMutations';
+import { EditDataType } from '../../profile/helpers/editData';
 
-type ElementSelector = {
+export type ElementSelector = {
   testId?: string;
   text?: string;
   valueSelector?: string;
+  id?: string;
+  querySelector?: string;
+  label?: string;
+};
+
+export type RenderHookResultsChildren = {
+  children: React.ReactNodeArray;
+};
+
+export type WaitForElementAndValueProps = {
+  selector: ElementSelector;
+  value?: string;
 };
 
 export type TestTools = RenderResult & {
   getElement: (selector: ElementSelector) => HTMLElement | null;
   waitForIsComplete: () => Promise<void>;
-  waitForDataChange: () => Promise<void>;
+  waitForDataChange: (previousChangeTime?: number) => Promise<void>;
   waitForElement: (selector: ElementSelector) => Promise<void>;
   getTextOrInputValue: (
     selector: ElementSelector
   ) => Promise<string | undefined>;
   fetch: () => Promise<void>;
+  triggerAction: (selector: ElementSelector) => Promise<void>;
+  submit: (props?: {
+    waitForOnSaveNotification?: WaitForElementAndValueProps;
+    waitForAfterSaveNotification?: WaitForElementAndValueProps;
+    skipDataCheck?: boolean;
+  }) => Promise<void>;
+  isDisabled: (element: HTMLElement | Element | null) => boolean;
+  setInputValue: ({
+    target,
+    selector,
+    newValue,
+  }: {
+    target?: HTMLElement;
+    selector: ElementSelector;
+    newValue: string;
+  }) => Promise<void>;
+  waitForElementAndValue: (props: WaitForElementAndValueProps) => Promise<void>;
+};
+
+export const cleanComponentMocks = (): void => {
+  cleanup();
+  cleanupHooks();
+  resetApolloMocks();
 };
 
 export const emptyResponseProvider: ResponseProvider = () => ({});
```

Comment:
Off-topic comment for line 90: the renderProfileContextWrapper seems to do more than just rendering; it also provides various test utility functions. Can't figure for a better name though. Maybe it is doing too much?

Reply:
Renamed to renderComponentWithMocksAndContexts. The return type in TS-files should be enough to tell what a function returns instead of using renderComponentWithMocksAndContextsAndReturnTestTools 😄 

All tested components need Router, mocked GraphQL, Profile Context, and Toast Context. And ability to fetch data, find elements, interact with DOM and waitFor something to be complete. This test helper returns context and tools for those. 

Plan B would be to render components in one func and pass result of that to another function(s) to do the rest. In each test file.